### ISSUE TITLE
Deceased Villager Data

### ISSUE THREAD

**Issue by BURNINGSHADOW278** [Luke100000/minecraft-comes-alive#978]
I'll start of saying this is potentially more a question than a Feature request, though I did not want to submit it as a bug so hence I chose this option.

**Is your feature request related to a problem? Please describe.**
The problem I'm having is getting back a deceased villager, who never got a gravestone due to either a bug or a glitch.
Though I suppose it also applies to Villagers whose gravestone has simply been destroyed.

**Describe the solution you'd like**
Is there any way, to get back a Gravestone for that villager or the Villager itself/entitydata. The villager exists in the Family Tree book, and I know the name of the villager. Is the villager data simply lost forever or is there a way to bring it back? If there currently isn't "that's okay" though I suppose it'd be nice if there was some way or command to get a gravestone/entitydata for a deceased villager. 

**Describe alternatives you've considered**
I've tried basically "getting the grave stone item" but although any gravestone from a villager only claims to have 2 NBT tags, in reality when I check it, it contains all of the villager's Entity data, so at least to me it didn't seem like there was any easy way to simply spawn in a gravestone with the villager's name to bring it back.

**Additional context**
Due to a glitch or otherwise there's ways "important" villagers can die without leaving or having a gravestone. My friend experienced a glitch where his "child" died without leaving a gravestone. But for example if a villager is petrified using the Fire and Ice Gorgon head, they will neither leave a gravestone, nor be considered deceased.
I've personally found no way to fix this or bring them back somehow anyway.
So if the entity data is truly gone/deleted that's okay, but I'm mostly wondering whether that's the case or whether it still exists but I just can't access it or don't know how to as the villager name and UUID seemingly still exist as it's in the Family Tree book.

**Comment by Luke100000** [comment]
Yes, the gravestone is where villager data is stored, other solutions would be too verbose. If you can reconstruct a way to cause the grave spawning to fail (where a villager has >10 hearts or has an active relationship with the player) I can try to fix it.

**Comment by BURNINGSHADOW278** [comment]
Alright, thank you for your quick reply, I figured as much but thanks for confirming it. If I can find any particular reason that caused this I'll be sure to come back. But I wasn't there when it happened though apparently a creeper blew up the villager, I tried in single player and that does actually seem reconstructible with the current mods we are using at least. I debugged a villager to +100 heart blew up a creeper next to it, it died and no gravestone appeared. Then I tried a second time making sure to talk to the villager after adding the 100 hearts to make sure it registered. That time the bone of the villager did drop suggesting that the same explosion from the creeper that killed the villager destroyed it's gravestone. (obviously turning off mob grieving would fix that, still even so most player gravestone mods wouldn't be destroyed by the explosion that killed the player only potentially a secondary explosion so perhaps there is a way to make it less likely to happen.)
Biggest difference from my tests was that my friend said he did not get the bone of his child/villager after the death/creeper explosion.
The only other case which is easy to reconstruct would be petrifying a villager using Fire & Ice's Gorgon Head. But honestly not sure if that needs "fixing" it feels like the intended/logical result.
